File: PowerOfThor1/PowerOfThor1.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>

int main()
{
    int lightX; // the X position of the light of power
    int lightY; // the Y position of the light of power
    int initialTX; // Thor's starting X position
    int initialTY; // Thor's starting Y position
    std::cin >> lightX >> lightY >> initialTX >> initialTY; std::cin.ignore();

    int thorPosX = initialTX;
    int thorPosY = initialTY;
    
    // game loop
    while (1) {
        int remainingTurns; // The remaining amount of turns Thor can move. Do not remove this line.
        std::cin >> remainingTurns; std::cin.ignore();

        std::string moveDir = "";

        if(thorPosY > lightY){ moveDir += "N"; thorPosY -= 1; } 
        else if(thorPosY < lightY){ moveDir += "S"; thorPosY += 1; }
        
        
        if(thorPosX > lightX){ moveDir += "W"; thorPosX -= 1; } 
        else if(thorPosX < lightX){ moveDir += "E"; thorPosX += 1; }
        
        
        std::cout << moveDir << std::endl;
    }
}

File: Easy/PowerOfThor1/PowerOfThor1.py
import sys
import math

light_x, light_y, initial_tx, initial_ty = [int(i) for i in input().split()]

thor_pos_x = initial_tx
thor_pos_y = initial_ty

while True:
    remaining_turns = int(input())  # The remaining amount of turns Thor can move. Do not remove this line.

    move_dir = ""

    if(thor_pos_y > light_y):
        move_dir += "N"
        thor_pos_y -= 1
        
    else:
        if(thor_pos_y < light_y):
            move_dir += "S"
            thor_pos_y += 1
        
        
    if(thor_pos_x > light_x):
        move_dir += "W"
        thor_pos_x -= 1
    else:
        if(thor_pos_x < light_x):
            move_dir += "E"
            thor_pos_x += 1

    print(move_dir)
